### PR TITLE
fix: array index typescript issues

### PR DESCRIPTION
**Pull request by amritk**
references #3769

This is a lot to do all at once, will chip away at it

## Release Notes

- **New Features**
  - Introduced a new type safety rule, `noUncheckedIndexedAccess`, enhancing type safety across the project.
  - Added a new TypeScript file, `node.ts`, to generate OpenAPI schema objects from TypeScript AST nodes.

- **Improvements**
  - Enhanced error handling and control flow in various functions to prevent runtime errors.
  - Simplified logic in several components for better readability and maintainability.

- **Bug Fixes**
  - Improved robustness in handling potentially undefined values across multiple files.

- **Documentation**
  - Updated import statements to clarify type-only imports, enhancing clarity and potentially reducing bundle size.

- **Tests**
  - Added new test cases for response generation, ensuring compliance with expected structures.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -3,16 +3,20 @@
     "target": "ES2022" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
     "module": "ESNext" /* Specify what module code is generated. */,
     "moduleResolution": "Bundler",
+    "moduleDetection": "force",
```
Comment (amritk) [inline tsconfig.json:6]: https://www.totaltypescript.com/tsconfig-cheat-sheet

Comment (tmastrom) [reply tsconfig.json:6]: We love Matt Pocock 🫡

**Review by tmastrom — COMMENTED** [review]
Heroic work in the tsconfig mines!

File: packages/code-highlight/src/code/line-numbers.ts [hunk 2]
```diff
@@ -42,12 +42,13 @@ export function codeBlockLinesPlugin() {
           // Adds a line break to the end of each line
           node.children.forEach((child: ElementContent) => {
             if (child.type === 'element' && child.tagName === 'span') {
-              const lastChild: ElementContent =
+              const lastChild: ElementContent | undefined =
```
Comment (hwkr) [inline packages/code-highlight/src/code/line-numbers.ts:45]: I wish you could use a question mark for this 😢

**Review by hwkr — APPROVED** [review]
Looks good to me! Should we have CodeRabbit have a look at it?

File: packages/code-highlight/src/code/line-numbers.ts (hunk 2 above)
Comment (amritk) [reply packages/code-highlight/src/code/line-numbers.ts:45]: whats the magic word

File: packages/draggable/src/Draggable.vue [hunk 3]
```diff
@@ -61,10 +61,8 @@ const emit = defineEmits<{
 }>()
 
 // The latest parentId in the arr should be the current parent
-const parentId = computed(() =>
-  (props.parentIds.length ?? 0) > 0
-    ? props.parentIds[props.parentIds.length - 1]
-    : null,
+const parentId = computed(
+  () => props.parentIds[props.parentIds.length - 1] ?? null,
```
[the remark this answers is not in the document]
Comment (amritk) [reply packages/draggable/src/Draggable.vue:65]: wow didn't even know this one

[the remark this answers is not in the document]
Comment (hwkr) [reply packages/draggable/src/Draggable.vue]: The future ✨

File: packages/code-highlight/src/code/line-numbers.ts (hunk 2 above)
Comment (hwkr) [reply packages/code-highlight/src/code/line-numbers.ts:45]: I don't think you can though right? Like that's not a typescript feature (I just wish it was)

**Review by marclave — APPROVED** [review]
nice fix : )

**Pull request merged**